salt/modules/gcp_addon.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+'''
+A route is a rule that specifies how certain packets should be handled by the
+irtual network. Routes are associated with virtual machine instances by tag,
+and the set of routes for a particular VM is called its routing table.
+For each packet leaving a virtual machine, the system searches that machine's
+routing table for a single best matching route.
+
+This module will create a route to send traffic destined to the Internet
+through your gateway instance.
+
+:codeauthor: :email:`Pratik Bandarkar <pratik.bandarkar@gmail.com>`
+:maturity:   new
+:depends:    google-api-python-client
+:platform:   Linux
+'''
+from __future__ import absolute_import
+import logging
+log = logging.getLogger(__name__)
+
+try:
+    import googleapiclient.discovery
+    import oauth2client.service_account
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
+
+
+def __virtual__():
+    '''
+    Check for googleapiclient api
+    '''
+    if HAS_LIB is False:
+        log.info("Required google API's(googleapiclient, oauth2client) not found")
+    return (HAS_LIB, "Required google API's(googleapiclient, oauth2client) not found")
+
+
+def _get_network_selfLink(project_id, network_name, service):
+    '''
+    Fetch network selfLink from network name.
+    '''
+
+    request = service.networks().get(project=project_id,
+                                     network=network_name).execute()
+    return request['selfLink']
+
+
+def _get_instance_selfLink(project_id, instance_name, instance_zone, service):
+    '''
+    Fetch instance selfLink from instance name.
+    '''
+
+    request = service.instances().get(project=project_id, zone=instance_zone,
+                                      instance=instance_name).execute()
+    return request['selfLink']
+
+
+def add_route(**kwargs):
```

Comment:
Since `kwargs` isn't being passed along to any other function call here, we would prefer if these keyword arguments were specifically listed in the interface.

Reply:
removed `kwargs` and updated the code.